harpy/validation/sam.py

```
@@ -50,4 +56,14 @@ def __init__(self, filenames):
                 "Files with clashing names",
                 dupe_out
             )
-
+        if detect_bc:
+            for i in range(min(6, self.count)):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

(B007)